components/Wizard.vue

```diff
@@ -0,0 +1,308 @@
+<script>
+/*
+Wizard accepts an array of steps (see props), and creates named slots for each step.
+It also contains slots for buttons:
+    next
+    cancel
+    finish
+
+Wizard will emit these events:
+    next({step})
+    cancel
+    finish
+*/
+
+export default {
+  props:      {
+    mode: {
+      type:    String,
+      default: 'create'
+    },
+
+    /*
+  steps need: {
+    name: String - this will be the slot name
+    label: String - this will appear in the top nav bar below circles
+    subText: String (optional) - If defined, appears in the Step {number}: banner. If blank, label is used
+    ready: Boolean - whether or not the step is completed/wizard is able to go to next step
+      if a step has ready=true, the wizard also allows navigation *back* to it
+  }
+  */
+    steps: {
+      type:    Array,
+      default: () => []
+    },
+
+    // Initial step to show when Wizard loads. This is overruled by the 'step' query param, if available
+    initStepIdx: {
+      type:    Number,
+      default: 0
+    },
+
+    // if true, allow users to navigate back to the first step of the Wizard
+    // if false, only way back to step 1 is to cancel and undo all configuration
+    editFirstStep: {
+      type:    Boolean,
+      default: false
+    }
+  },
+
+  asyncData(ctx) {
+    const { route:{ query } } = ctx;
+
+    return { queryStep: query.step };
+  },
```

Comment:
Components don't have an `asyncData`, so this can't work?  But you can do this + what `data` does in `fetch`